lib/Basic/Version.cpp

```diff
@@ -285,6 +285,16 @@ Version::preprocessorDefinition(StringRef macroName,
   return define;
 }
 
+bool
+Version::isValidEffectiveLanguageVersion() const
+{
+  // At the moment, support setting Swift effective language version to
+  // any version starting with major number 3 or 4.
+  return Components.size() > 0 &&
+    (Components[0] == 3 ||
+     Components[0] == 4);
```

Comment:
Ok. I put in support for 4.x as a sort of cheap means to enable testing that `-swift-version` actually _does_ change the version (since current == 3.0 already) but I will change this to a shorter whitelist and weaken the test to just confirm that `-swift-version 3` keeps working when we _do_ bump the language version.